src/main/java/org/elasticsearch/index/translog/Translog.java

```diff
@@ -158,6 +160,53 @@
      */
     long findLargestPresentTranslogId() throws IOException;
 
+    /**
+     * Returns an OperationIterator to iterate over all translog entries in the given translog ID.
+     * @throws java.io.FileNotFoundException if the file for the translog ID can not be found
+     */
+    OperationIterator openIterator(long translogId) throws IOException;
+
+    /**
+     * Iterator for translog operations.
+     */
+    static class OperationIterator implements Closeable {
```

Comment:
you can take over taht if you want I think translog API is broken anyway so we ahve to overhault it - this makes things better than worse